charts/opentelemetry-operator/templates/deployment.yaml

```diff
@@ -23,7 +23,7 @@ spec:
       hostNetwork: {{ .Values.hostNetwork }}
       containers:
         - args:
-            - --metrics-addr=127.0.0.1:{{ .Values.manager.ports.metricsPort }}
+            - --metrics-addr=0.0.0.0:{{ .Values.manager.ports.metricsPort }}
```

Comment:
Is feels like it could be a breaking change.  If the service monitor needs `0.0.0.0` instead of `127.0.0.1` then it needs to be configurable based on whether or not the service monitor is enabled.

Reply:
0.0.0.0 includes 127.0.0.1 so I don't think it will be a breaking change. (Tested with a sidecar container hitting 127.0.0.1 on this branch)

Reply:
Thanks for doing that test. @dmitryax do you want to do this as part of this PR or should it be separate?

Reply:
Looks like this PR has many unrelated changes. Let's break them down.

This particular change can be considered insecure. I would say we should keep 127.0.0.1 by default and has an option to override it?

Reply:
I don't really know from the top of my head. The operator chart was introduced by someone who is not involved in the project anymore. We are actively looking for code owners for the operator helm chart. Maybe @open-telemetry/operator-approvers can answer your question